src/app/new-courses-page/course-date/course-date.component.ts

```diff
@@ -0,0 +1,17 @@
+import { ChangeDetectionStrategy, Component, EventEmitter, Input, Output } from '@angular/core';
+
+@Component({
+  selector: 'app-course-date',
+  templateUrl: './course-date.component.html',
+  styleUrls: ['./course-date.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class CourseDateComponent {
+  @Input() courseDate: Date;
+  @Output() courseDateChange = new EventEmitter<Date>();
+
+  onDateChange(model: Date): void {
+    this.courseDate = model;
```

Comment:
Do not change the input. It should be changed by the binding in the parent component. That's called unidirectional data flow
https://blog.angular-university.io/angular-2-what-is-unidirectional-data-flow-development-mode/

Reply:
Changed. Thanks)